tests/letstest/scripts/test_sdists.sh

```diff
@@ -12,6 +12,8 @@ export VENV_ARGS="-p $PYTHON"
 # setup venv
 tools/_venv_common.py --requirement letsencrypt-auto-source/pieces/dependency-requirements.txt
 . ./venv/bin/activate
+# pytest is needed to run tests on some of our packages so we install a pinned version here.
+tools/pip_install.py pytest
```

Comment:
Why not pin pytest for all of our installations? Presumably an older version is getting installed over on line 13?

Reply:
We have `pytest` pinned in [dev_constraints.txt](https://github.com/certbot/certbot/blob/bb1242c04767c8686de39af78f4f562ae33ded1d/tools/dev_constraints.txt#L50) which `tools/pip_install.py` is using. In general, as long as we use our scripts in `tools`, everything is pinned down (unless you set the environment variable `CERTBOT_NO_PIN`). It isn't included in certbot-auto's dependencies being used on line 13 though because it doesn't need to be installed when we're not running tests.

We could potentially do something like install all of `dev_constraints.txt` here if you prefer. This will install a lot of unnecessary stuff and may mask problems if we incorrectly modify `setup.py` in the future (such as a dependency being missing or incorrectly specified in `setup.py` but included properly in our dev tools), but it should pretty robustly install everything we'd ever need.

Reply:
How was it getting installed before?

Reply:
`setuptools` pulled it in which you can see in the log I put in #6763.

Reply:
😱

Reply:
Ah, I figured it out. I was confused as to why this was only breaking here. Turns out we [pin pytest directly](https://github.com/certbot/certbot/blob/master/tests/letstest/scripts/test_tests.sh) in the only other test farm test to use it. We should probably match how we're doing it in both places?

Reply:
Nice catch! I agree we should take the same approach in both files. I added a commit making that change.

Reply:
I ran test_tests.sh test farm tests with this change and they all passed.